timesketch/api/v1/resources/upload.py

```diff
@@ -162,11 +163,24 @@ def _upload_and_index(
                 timeline = timeline_
                 break
 
-            abort(
-                HTTP_STATUS_CODE_BAD_REQUEST,
+            logger.error(
                 'There is a timeline in the sketch that has the same name '
-                'but is stored in a different index, check the data_label '
-                'on the uploaded data')
+                'but is stored in a different index: name {0:s} attempting '
+                'index: {1:s} but found index {2:s} - retrying with a '
+                'different timeline name.'.format(
+                    timeline_name, searchindex.index_name,
+                    timeline_.searchindex.index_name))
+
+            timeline_name = '{0:s}_{1:s}'.format(
+                timeline_name, uuid.uuid4().hex)
```

Comment:
This will become a very long name in the UI. Could we generate a shorter uniq string here?

Reply:
forgot to shorten this, the idea was to just use the last five characters.